contributing_to_docs/doc_guidelines.adoc

```diff
@@ -911,22 +911,134 @@ kind: Pod
 metadata:
   labels:
     test: liveness
-...
+# ...
 ----
 +
 or
 +
 ----
-...
+# ...
 spec:
   containers:
   - args:
     image: k8s.gcr.io/liveness
-...
+# ...
 ----
 +
 Do not use `[...]`, `<snip>`, or any other variant.
 
+* Do not use `[source,yaml]` for the following:
+
+** Bash "here" documents, such as the following example:
++
+....
+[source,terminal]
+----
+$ cat <<EOF| oc create -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mlistener
+  labels:
+    app: multicast-verify
+EOF
+----
+....
+
+** The output of `oc describe <pural> <object>`, such as in the following example:
++
+....
+[source,text]
+----
+Name:               node1.example.com
+Roles:              worker
+Labels:             beta.kubernetes.io/arch=amd64
+...
+Annotations:        cluster.k8s.io/machine: openshift-machine-api/ahardin-worker-us-east-2a-q5dzc
+...
+CreationTimestamp:  Wed, 13 Feb 2019 11:05:57 -0500
+----
+....
+
+
+=== YAML formatting for Kubernetes and OpenShift API objects
+The following formatting guidelines apply to YAML manifests, but do not apply to the installation configuration YAML specified by `install-config.yaml`.
+
+==== Required fields
+
+- Always include the `apiVersion` and `kind` so that a user always knows the context of the YAML.
+- Always include the full hierarchy to a deeply nested key.
+- For objects that are in the global scope, such as for `config.openshift.io` API group, always include the `metadata.name` for the object, which is usually `cluster`.
```

Comment:
Just wondering if the metadata.name should be required for all objects. I assume it's a required field, just like apiversion and kind are.

Reply:
@bergerhoffer, it is, but the value is generally user-provided in other cases, and I don't know if we want to require that every YAML always have:

```yaml
metadata:
  name: <name> (1)

(1) Specify the name of the object.
```
Or similar.

I thought writers might want to be able to continue to waive that away with:

```yaml
apiVersion: someApi
kind: baz
# ...
spec:
  key: "value"
```